projects/step-frontend/src/lib/modules/timeseries/modules/_common/types/time-series/time-series-sync-group.ts

```diff
@@ -0,0 +1,87 @@
+import { TimeSeriesKeywordsContext } from './time-series-keywords.context';
+import { Observable, Subject } from 'rxjs';
+
+export class TimeSeriesSyncGroup {
+  readonly id: string; // usually the id of the master chart
+
+  seriesVisibility: Record<string, boolean> = {};
+
+  allSeriesShow$ = new Subject<void>();
```

Comment:
Internal subjects are visible from outside.
Also they are not completed

```suggestion
  private allSeriesShowInternal$ = new Subject<void>();
  readonly allSeriesShow$ = this.allSeriesShowInternal$.asObservable();
```

Reply:
@dvladir I made them private but I'd definitely like to have a method instead

Reply:
Context, sync group, colorsPool all use the same 'onBlablaChange()' approach - it's more clear imo